target_chains/ethereum/contracts/forge-test/utils/PythTestUtils.t.sol

```diff
@@ -399,4 +410,44 @@ contract PythUtilsTest is Test, WormholeTestUtils, PythTestUtils, IPythEvents {
         // 0.00001 and we are targetDecimals is 2.
         assertEq(PythUtils.convertToUint(100, -5, 2), 0);
     }
+
+    function testCombinePrices() public {
+
+        // Basic Tests 
+        successTest(100, -2, 100, -2, 100, -2);
+        successTest(10000, -2, 100, -2, 10000, -2);
+        successTest(1_000_000, -2, 10_000, -2, 10_000, -2);
+
+        // Negative Price Tests
+        revertTest(-100, -2, 100, -2);
+        revertTest(100, -2, -100, -2);
+        revertTest(-100, -2, -100, -2);
+
+        // Positive Exponent Tests
+        revertTest(100, 2, 100, -2);
+        revertTest(100, -2, 100, 2);
+        revertTest(100, 2, 100, 2);
+
+        // Different Exponent Tests
+        successTest(10_000, -2, 100, -4, 100_000_000, -4);
+        successTest(10_000, -2, 10_000, 0, 1, -2);
+        successTest(10_000, 0, 10_000, 0, 1, 0);
+
+        // End Range Tests
+        successTest(int64(type(int64).max), 0, int64(type(int64).max), 0, 1, 0);
+        successTest(int64(type(int64).max), 0, 1, 0, int64(type(int64).max), 0);
+        successTest(1, 0, int64(type(int64).max), 0, 1 / int64(type(int64).max), 0);
```

Comment:
this case is the kind of thing you have to worry about with these conversion methods -- the `1 / int64(...)` here is actually `0`, which could be disastrous depending on the use case.

The critical thing to do here is to make sure that users understand *when* they can hit this case. I think what your method does is truncate (ie. round down) to the nearest representable quantity in the smaller exponent. Please document that explicitly. You may also want to let callers pass in the exponent for the result -- it's very easy to be safe if the caller explicitly specifies `-8` or something similar as the exponent.

Reply:
I planned to ask user to send an exponent for more precision, but what should be the limit for that? Should we cap it to 18?